packages/core/src/odata/v2/request-builder/update-request-builder.ts

```diff
@@ -91,20 +102,45 @@ export class UpdateRequestBuilderV2<EntityT extends EntityV2>
 
     return (
       this.build(destination, options)
+        .then(request => this.warnIfNavigation(request))
         .then(request => request.execute())
 
         // Update returns 204 hence the data from the request is used to build entity for return
-        .then(response =>
-          this._entity
+        .then(response => {
+          const eTag =
+            extractEtagFromHeader(response.headers) ||
+            extractODataEtagV2(response.data) ||
+            this.requestConfig.eTag;
+          return this._entity
             .setOrInitializeRemoteState()
-            .setVersionIdentifier(this.requestConfig.eTag)
-        )
+            .setVersionIdentifier(eTag);
+        })
         .catch(error =>
           Promise.reject(errorWithCause('OData update request failed!', error))
         )
     );
   }
 
+  /*
+   * In case the entity contains a navigation to a different entity a warning is printed.
+   */
+  warnIfNavigation(
```

Comment:
I don't think this should be a instance method, especially not a public one.

Reply:
true. I make it not visible and see about the scope.